cluster/images/kube-discovery/README.md

```diff
@@ -0,0 +1,39 @@
+### kube-discovery
+
+An initial implementation of a Kubernetes discovery service using JSON Web Signatures.
+
+This prototype is configured by kubeadm and run within kubernetes itself.
+
+## Requirements
+
+Generate a CA cert save it to: /tmp/secret/ca.pem to run the service or unit tests. (will not be required for unit tests for long) Similarly when run within kubernetes we expect a secret to be provided at this location as well. (see below)
+
+## Build And Run From Source
+
+```
+$ build/run.sh /bin/bash -c "KUBE_BUILD_PLATFORMS=linux/amd64 make WHAT=cmd/kube-discovery"
+$ _output/dockerized/bin/linux/amd64/kube-discovery
+2016/08/23 19:17:28 Listening for requests on port 9898.
+
+```
+
+## Running in Docker
+
+This image is published temporarily on Docker Hub as dgoodwin/kubediscovery
+
+`docker run --rm -p 9898:9898 -v /tmp/secret/ca.pem:/tmp/secret/ca.pem --name kubediscovery dgoodwin/kubediscovery`
+
+## Running in Kubernetes
+
+A dummy certificate is included in ca-secret.yaml.
+
+```
+kubectl create -f ca-secret.yaml
+kubectl create -f kubediscovery.yaml
+```
+
+## Testing the API
+
+`curl "http://localhost:9898/cluster-info/v1/?token-id=TOKENID"`
```

Comment:
Can you define what a valid token is and where it comes from. To me, it's not obvious here.
But I do not want to block on docs like this initially...